src/main/java/org/team1540/robot2019/datastructures/utils/Transform2DRollingAverage.java

```diff
@@ -0,0 +1,41 @@
+package org.team1540.robot2019.datastructures.utils;
+
+import static java.lang.Math.atan2;
+import static java.lang.Math.cos;
+import static java.lang.Math.sin;
+
+import com.google.common.collect.EvictingQueue;
+import java.util.Queue;
+import org.team1540.robot2019.datastructures.twod.Transform2D;
+
+public class Transform2DRollingAverage {
+
+    private Queue<Transform2D> poseQueue;
+
+    public Transform2DRollingAverage(int numElements) {
+        this.poseQueue = EvictingQueue.create(numElements);
```

Comment:
I think WPILib also has a `RingBuffer` class which might be more efficient

Reply:
Cool, the evicting queue also works though.